forc-plugins/forc-client/src/ops/deploy/op.rs

```diff
@@ -14,8 +14,10 @@ use tracing::info;
 
 use crate::ops::{deploy::cmd::DeployCommand, parameters::TxParameters};
 
-/// Checks if we have a WorkspaceManifest and if that is the case we will be deploying the packages
-/// listed in the members field of the WorkspaceManifest in the order of their declaration
+/// If the provided path (`DeployCommand.path`, or if it is not provided the current directory)
+/// contains a workspace manifest file, `deploy` will be deploying the workspace members in the
+/// order that their listing in the manifest file. If the provided path contains only a package
```

Comment:
> `deploy` will be deploying the workspace members in the order that their listing in the manifest file

I think we'll need to make sure we actually deploy these in order of dependency, which we can get by doing a toposort of the subgraph of all `contract-dependencies` filtered down to those that exist in our workspace members set. Otherwise if we try to deploy a contract before one of its dependencies is deployed we'll run into problems :) We could require the user to specify these in order, but it would be nice if this weren't necessary.

I think this graph of contract dependencies is something that could be returned from the result of the `forc_pkgs::build` functions, so it might make sense to first address #2931 and get contract dependencies into the pkg graph before addressing this.